NitroxServer/GameLogic/Unlockables/StoryGoalData.cs

```diff
@@ -18,9 +19,18 @@ public class StoryGoalData
         [JsonProperty, ProtoMember(3)]
         public ThreadSafeCollection<string> GoalUnlocks { get; } = new ThreadSafeCollection<string>();
 
-        public void RemovedLatestRadioMessage()
+        public bool RemovedLatestRadioMessage()
         {
-            RadioQueue.RemoveAt(0);
+            if (RadioQueue.Count > 0)
+            {
+                RadioQueue.RemoveAt(0);
+                return true;
+            }
+            else
+            {
+                Log.Error("Couldn't remove the latest radio message: the radio queue is empty");
```

Comment:
You're logging twice for same issue so this log can be moved to the RadioPlay.Process instead (and adding the packet info to it as well).